Error: `FAILED midi_app_controller/gui/_tests/test_midi_status.py::test_edit_binds - AttributeError: 'NoneType' object has no attribute 'name'`

<details>
<summary>Results from local testing on macOS Tahoe</summary>

```
================================== test session starts ====================================
platform darwin -- Python 3.13.7, pytest-9.0.3, pluggy-1.6.0
PyQt6 6.11.0 -- Qt runtime 6.11.0 -- Qt compiled 6.11.0
rootdir: /Users/willingc/Code/repos-napari/midi-app-controller
configfile: pyproject.toml
plugins: zarr-3.1.6, cov-7.1.0, napari-0.7.0, qt-4.5.0, napari-plugin-engine-0.2.1, npe2-0.8.2
collected 209 items                                                                        

midi_app_controller/_tests/test_config.py ...                                        [  1%]
midi_app_controller/_tests/test_utils.py ........                                    [  5%]
midi_app_controller/actions/_tests/test_actions_handler.py .................         [ 13%]
midi_app_controller/actions/_tests/test_bound_controller.py .....................    [ 23%]
midi_app_controller/actions/_tests/test_napari_actions.py ............               [ 29%]
midi_app_controller/controller/_tests/test_connected_controller.py ................. [ 37%]
.............                                                                        [ 43%]
midi_app_controller/gui/_tests/test_binds_editor.py ................................ [ 58%]
...                                                                                  [ 60%]
midi_app_controller/gui/_tests/test_midi_status.py ...F......                        [ 65%]
midi_app_controller/models/_tests/test_binds.py ........                             [ 68%]
midi_app_controller/models/_tests/test_controller.py .............                   [ 75%]
midi_app_controller/models/_tests/test_utils.py .........                            [ 79%]
midi_app_controller/state/_tests/test_state_manager.py ............................. [ 93%]
..............                                                                       [100%]

========================================= FAILURES =========================================
_____________________________________ test_edit_binds ______________________________________

binds_editor_mock = <MagicMock name='BindsEditor' id='5224930608'>
midi_status = <midi_app_controller.gui.midi_status.MidiStatus object at 0x1233816d0>
patch_rtmidi = ([SelectedItem(name='Binds 1', path=PosixPath('/path/binds_1.yaml')), SelectedItem(name='Binds 2', path=PosixPath('/pa...h=PosixPath('/path/controller_2.yaml')), SelectedItem(name='Controller 3', path=PosixPath('/path/controller_3.yaml'))])
qtbot = <pytestqt.qtbot.QtBot object at 0x123291e90>

    @patch("midi_app_controller.gui.midi_status.BindsEditor")
    def test_edit_binds(binds_editor_mock, midi_status, patch_rtmidi, qtbot):
        binds_names, controller_names = patch_rtmidi
        state_manager = get_state_manager()
        QTest.qWait(500)
        qtbot.keyClicks(midi_status.current_binds, "(none)")
    
        for i in range(len(binds_names)):
            QTest.qWait(500)
            qtbot.keyClicks(midi_status.current_binds, binds_names[i].name)
            QTest.qWait(500)
    
>           assert state_manager.selected_binds.name == binds_names[i].name
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E           AttributeError: 'NoneType' object has no attribute 'name'

midi_app_controller/gui/_tests/test_midi_status.py:165: AttributeError
------------------------------------ Captured log call -------------------------------------
ERROR    root:connected_controller.py:287 Value Error: 
ERROR    root:connected_controller.py:287 Value Error: 
ERROR    root:connected_controller.py:287 Value Error:
===================================== warnings summary =====================================
midi_app_controller/models/binds.py:65
  /Users/willingc/Code/repos-napari/midi-app-controller/midi_app_controller/models/binds.py:65: PydanticDeprecatedSince212: Using `@model_validator` with mode='after' on a classmethod is deprecated. Instead, use an instance method. See the documentation at https://docs.pydantic.dev/2.12/concepts/validators/#model-after-validator. Deprecated in Pydantic V2.12 to be removed in V3.0.
    @model_validator(mode="after")

midi_app_controller/models/controller.py:68
  /Users/willingc/Code/repos-napari/midi-app-controller/midi_app_controller/models/controller.py:68: PydanticDeprecatedSince212: Using `@model_validator` with mode='after' on a classmethod is deprecated. Instead, use an instance method. See the documentation at https://docs.pydantic.dev/2.12/concepts/validators/#model-after-validator. Deprecated in Pydantic V2.12 to be removed in V3.0.
    @model_validator(mode="after")

midi_app_controller/models/controller.py:99
  /Users/willingc/Code/repos-napari/midi-app-controller/midi_app_controller/models/controller.py:99: PydanticDeprecatedSince212: Using `@model_validator` with mode='after' on a classmethod is deprecated. Instead, use an instance method. See the documentation at https://docs.pydantic.dev/2.12/concepts/validators/#model-after-validator. Deprecated in Pydantic V2.12 to be removed in V3.0.
    @model_validator(mode="after")

midi_app_controller/models/controller.py:107
  /Users/willingc/Code/repos-napari/midi-app-controller/midi_app_controller/models/controller.py:107: PydanticDeprecatedSince212: Using `@model_validator` with mode='after' on a classmethod is deprecated. Instead, use an instance method. See the documentation at https://docs.pydantic.dev/2.12/concepts/validators/#model-after-validator. Deprecated in Pydantic V2.12 to be removed in V3.0.
    @model_validator(mode="after")

.venv/lib/python3.13/site-packages/pydantic/_internal/_generate_schema.py:319: 16 warnings
  /Users/willingc/Code/repos-napari/midi-app-controller/.venv/lib/python3.13/site-packages/pydantic/_internal/_generate_schema.py:319: PydanticDeprecatedSince20: `json_encoders` is deprecated. See https://docs.pydantic.dev/2.12/concepts/serialization/#custom-serializers for alternatives. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.12/migration/
    warnings.warn(

midi_app_controller/gui/_tests/test_binds_editor.py: 15 warnings
  /Users/willingc/Code/repos-napari/midi-app-controller/midi_app_controller/gui/binds_editor.py:453: PydanticDeprecatedSince20: The `copy` method is deprecated; use `model_copy` instead. See the docstring of `BaseModel.copy` for details about how to handle `include` and `exclude`. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.12/migration/
    self.binds = binds.copy(deep=True)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================= short test summary info ==================================
FAILED midi_app_controller/gui/_tests/test_midi_status.py::test_edit_binds - AttributeError: 'NoneType' object has no attribute 'name'
======================= 1 failed, 208 passed, 35 warnings in 25.41s ========================
```
</details>

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -10,6 +10,7 @@ jobs:
   run:
     runs-on: ubuntu-latest
     strategy:
+      fail-fast: false
       matrix:
         python-version: ['3.10', '3.11', '3.12', '3.13', '3.14']
     env:

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -8,13 +8,12 @@ on:
 
 jobs:
   run:
-    runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
         python-version: ['3.10', '3.11', '3.12', '3.13', '3.14']
-    env:
-      DISPLAY: ':99.0'
+        os: [ubuntu-latest]
+    runs-on: ${{ matrix.os }}
     steps:
       - name: Checkout
         uses: actions/checkout@de0fac2e4500dabe0009e67214ff5f5447ce83dd # v6.0.2
@@ -25,6 +24,10 @@ jobs:
           python-version: ${{ matrix.python-version }}
           cache: 'pip'
           cache-dependency-path: pyproject.toml
+
+      - name: Install alsa deps
+        if: matrix.os == 'ubuntu-latest'
+        run: sudo apt-get install libasound2-dev
 
       - name: Setup headless display
         uses: pyvista/setup-headless-display-action@5bc8de3bc71fcda7a96439571287a554901541a0 # v4.3

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -25,13 +25,11 @@ jobs:
           cache: 'pip'
           cache-dependency-path: pyproject.toml
 
-      - name: Install system dependencies (GUI + python-rtmidi / ALSA)
-        run: |
-          sudo apt update
-          sudo apt install -y \
-            libasound2-dev \
-            libxkbcommon-x11-0 libxcb-icccm4 libxcb-image0 libxcb-keysyms1 libxcb-randr0 libxcb-render-util0 libxcb-xinerama0 libxcb-xfixes0 x11-utils libgl1 libegl1 libdbus-1-3
-          /sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_99.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :99 -screen 0 1920x1200x24 -ac +extension GLX
+      - name: Setup headless display
+        uses: pyvista/setup-headless-display-action@5bc8de3bc71fcda7a96439571287a554901541a0 # v4.3
+        with:
+          qt: true
+          wm: herbstluftwm
 
       - name: Install dependencies
         run: pip install -e .[testing]

--- a/.gitignore
+++ b/.gitignore
@@ -1,3 +1,4 @@
+midi_app_controller/_version.py
 # Byte-compiled / optimized / DLL files
 __pycache__/
 *.py[cod]

